src/sonic-config-engine/portconfig.py

```diff
@@ -45,16 +46,23 @@
 #
 # Helper Functions
 #
+
+# For python2 compatibility
+def jsonStrHook(j):
```

Comment:
>jsonStrHook [](http://example.com/codeflow?start=4&length=11)

Why we need this hook for python2? Can we assume platform.json and hwsku.json do not contain chars not in ascii range?

If we really need this hook for python2, can we add it only for python2? Python3 just has no hook at all.

Reply:
@qiluo-msft 
The only reason I added py2 compatibility is failing unit tests for sonic-config-engine py2 package. These are built because ENABLE_PY2_MODULES is yes for buster and jessie. Any reason to keep it as of now?

Regarding non-ascii chars Idk. It's so happens we have an implicit requirement to not have bools which we need for sure. Plus I have not found any platform.json in master having non-ascii. So currently I cannot think of a non-compromise solution. Ideally we just drop py2.

Reply:
If we really need this hook for python2, can we add it only for python2? Python3 just has no hook at all.

Reply:
Alternatively in json we get from json.load we can replace bool values to some string makers and then replace the markers back with False or True before passing to ast.literal_eval. This also would keep compatibility with non-ascii. 

@qiluo-msft But do we really need py2 compatibility? What are the plans for buster?

Reply:
@saiarcot895 Could you help answer the py2 plan on buster?

Reply:
Buster will continue to have Python 2.

However, from what I can tell, `rules/docker-config-engine-buster.mk` is using the python 3 version of sonic-config-engine, so this might not be needed on Buster containers. (I verified on a couple Buster containers that `sonic-cfggen` is calling the Python 3 interpreter.)

Stretch is using the python 2 version of sonic-config-engine. Looks like the syncd container for innovium and nephos are on Stretch.